### PR TITLE
GAPI: Add normalize kernel in G-API

### PR DESCRIPTION
**Pull request by smirnov-alexey**
This pullrequest adds normalize kernel and tests on it

```
buildworker:Win64 OpenCL=windows-2
```

### CODE REVIEW

File: modules/gapi/include/opencv2/gapi/imgproc.hpp [hunk 1]
```diff
@@ -154,6 +154,19 @@ namespace imgproc {
             return in.withType(CV_8U, 1);
         }
     };
+
+    G_TYPED_KERNEL(Gnormalize, <GMat(GMat, double, double,
+                    int, int)>, "org.opencv.imgproc.normalize") {
+        static GMatDesc outMeta(GMatDesc in, double, double,
+                    int, int ddepth) {
```
Comment (dmatveev) [inline modules/gapi/include/opencv2/gapi/imgproc.hpp]: Something really strange happened with indentation here.

File: modules/gapi/include/opencv2/gapi/imgproc.hpp [hunk 2]
```diff
@@ -154,6 +154,19 @@ namespace imgproc {
...
+            // unlike opencv doesn't have a mask as a parameter
+            if (ddepth < 0){
+                return in;
+            }else{
+                return in.withDepth(ddepth);
```
Comment (dmatveev) [inline modules/gapi/include/opencv2/gapi/imgproc.hpp]: Use `? :`

File: modules/gapi/include/opencv2/gapi/imgproc.hpp [hunk 3]
```diff
@@ -670,6 +683,30 @@ Output image must be 8-bit unsigned 3-channel image @ref CV_8UC3.
...
+normalization.
+@param beta upper range boundary in case of the range normalization; it is not used for the norm
+normalization.
+@param norm_type normalization type (see cv::NormTypes).
+@param dtype when negative, the output array has the same type as src; otherwise, it has the same
+number of channels as src and the depth =CV_MAT_DEPTH(dtype).
+@sa norm, Mat::convertTo, SparseMat::convertTo
+*/
+GAPI_EXPORTS GMat normalize(const GMat& src, double alpha, double beta,
+                    int norm_type, int ddepth);
```
Comment (dmatveev) [inline modules/gapi/include/opencv2/gapi/imgproc.hpp]: We usually align code so `int norm_type` comes at the same column as `const GMat &src` one line above

File: modules/gapi/test/common/gapi_imgproc_tests_inl.hpp [hunk 4]
```diff
@@ -625,6 +625,41 @@ TEST_P(YUV2BGRTest, AccuracyTest)
...
+    compare_f cmpF = std::get<0>(param);
+    const auto in_type = std::get<1>(param);
+    const auto in_size = std::get<2>(param);
+    const auto a = std::get<3>(param);
+    const auto b = std::get<4>(param);
+    const auto norm_type = std::get<5>(param);
+    const auto ddepth = std::get<6>(param);
+    const auto createOut = std::get<7>(param);
+    auto compile_args = std::get<8>(param);
+    int dtype = CV_MAKETYPE(ddepth, CV_MAT_CN(in_type));
```
Comment (dmatveev) [inline modules/gapi/test/common/gapi_imgproc_tests_inl.hpp]: What about `std::tie()`?

If a new parameter is either added or removed, you'd not need to fix all indexes in every `std::get<>`

File: modules/gapi/include/opencv2/gapi/imgproc.hpp [hunk 5]
```diff
@@ -154,6 +154,13 @@ namespace imgproc {
             return in.withType(CV_8U, 1);
         }
     };
+
+    G_TYPED_KERNEL(Gnormalize, <GMat(GMat, double, double, int, int)>, "org.opencv.imgproc.normalize") {
```
Comment (rgarnov) [inline modules/gapi/include/opencv2/gapi/imgproc.hpp]: Actually cv::normalize is defined in core.hpp, so cv::gapi::normalize should resize in gapi/core.hpp

File: modules/gapi/include/opencv2/gapi/imgproc.hpp [hunk 6]
```diff
@@ -670,6 +677,32 @@ Output image must be 8-bit unsigned 3-channel image @ref CV_8UC3.
...
 
+/** @brief Normalizes the norm or value range of an array.
+
+The function normalizes scale and shift the input array elements so that
+\f[\| \texttt{dst} \| _{L_p}= \texttt{alpha}\f]
+(where p=Inf, 1 or 2) when normType=NORM_INF, NORM_L1, or NORM_L2, respectively; or so that
+\f[\min _I  \texttt{dst} (I)= \texttt{alpha} , \, \, \max _I  \texttt{dst} (I)= \texttt{beta}\f]
+when normType=NORM_MINMAX (for dense arrays only).
+
+In case of sparse matrices, only the non-zero values are analyzed and transformed. Because of this,
```
Comment (rgarnov) [inline modules/gapi/include/opencv2/gapi/imgproc.hpp]: Is sparse matrix case related to our scope?

File: modules/gapi/src/backends/cpu/gcpuimgproc.cpp [hunk 7]
```diff
@@ -244,6 +244,15 @@ GAPI_OCV_KERNEL(GCPURGB2GrayCustom, cv::gapi::imgproc::GRGB2GrayCustom)
     }
 };
 
+GAPI_OCV_KERNEL(GCPUnormalize, cv::gapi::imgproc::Gnormalize)
```
Comment (rgarnov) [inline modules/gapi/src/backends/cpu/gcpuimgproc.cpp]: GCPUnormalize -> GCPUNormalize, Gnormalize -> GNormalize. (cv::gapi::normalize(..) should remain as is)

**Comment by dmatveev** [comment]
@smirnov-alexey please reply in open discussions when you address a review issue. Then it is easier for reviewers to review knowing which issue you already fixed and which not.

File: modules/gapi/src/api/kernels_core.cpp [hunk 8]
```diff
@@ -355,5 +355,11 @@ GMat sqrt(const GMat& src)
     return core::GSqrt::on(src);
 }
 
+GMat normalize(const GMat& _src, double a, double b,
+                    int norm_type, int ddepth)
```
Comment (dmatveev) [inline modules/gapi/src/api/kernels_core.cpp]: Alignment was not fixed

Comment (smirnov-alexey) [reply modules/gapi/src/api/kernels_core.cpp]: What do you mean? normalize() itself should remain in lower case

**Comment by dmatveev** [comment]
👍 thanks!

**Pull request merged**